service/controller/v3/network/network.go

```diff
@@ -0,0 +1,65 @@
+package network
+
+import (
+	"context"
+	"net"
+
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/service/controller/setting"
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+const (
+	ipv4Mask = 32
+)
+
+// Compute network subnets within network from CR.
+func ComputeFromCR(ctx context.Context, obj interface{}, networkSetting setting.AzureNetwork) (*Subnets, error) {
+	azureConfig, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	vnetCIDR := key.VnetCIDR(azureConfig)
+	_, vnet, err := net.ParseCIDR(vnetCIDR)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return Compute(*vnet, networkSetting)
```

Comment:
I said this already a couple of times. It is really helpful when we do not do these shortcuts because we do error masking and when the stack hides certain locations the stack does not match reality and the mental model of the one who has to debug something breaks and all gets just more complicated.

Reply:
fixed, sorry I was not 100% sure about this one.